kratos/utilities/transformation_utilities.cpp

```diff
@@ -0,0 +1,111 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Aditya Ghantasala
+//     Re-factor:    Vicente Mataix Ferrandiz
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "utilities/transformation_utilities.h"
+
+namespace Kratos
+{
+namespace TransformationUtilities
+{
+
+void CalculateTranslationMatrix(
+    const double Modulus,
+    MatrixType& rMatrix,
+    const DenseVector<double>& rDirOfTranslation
+    )
+{
+    rMatrix(0,0) = 1.0;
+    rMatrix(0,1) = 0.0;
+    rMatrix(0,2) = 0.0;
+    rMatrix(0,3) = Modulus * rDirOfTranslation[0];
+    rMatrix(1,0) = 0.0;
+    rMatrix(1,1) = 1.0;
+    rMatrix(1,2) = 0.0;
+    rMatrix(1,3) = Modulus * rDirOfTranslation[1];
+    rMatrix(2,0) = 0.0;
+    rMatrix(2,1) = 0.0;
+    rMatrix(2,2) = 1.0;
+    rMatrix(2,3) = Modulus * rDirOfTranslation[2];
+    rMatrix(3,0) = 0.0;
+    rMatrix(3,1) = 0.0;
+    rMatrix(3,2) = 0.0;
+    rMatrix(3,3) = 1.0;
```

Comment:
I think it would be less code if you assign an identity matrix and only modify the off-diagonal terms

Reply:
Then you are assigning twice some values...

Reply:
hm ok, up to you then